### DISCUSSION TITLE
Additional tagging with regexp- mutate model.two or post-process hook?

### DISCUSSION THREAD

**Discussion by track0x1** [nlp-compromise/nlp_compromise#1086] [Q&A] [answered]
I need to do some additional tagging and was wondering if there are pros/cons to mutating model.two by injecting my new regex into regexNormal versus adding a post-process compute hook that performs the additional tagging through `match().tag()`.

Code examples:
```js
// Mutate regexNormal
const plugin = {
  mutate: world => {
    const additionalTags = [
      [/^[\u00BC-\u00BE\u2150-\u215E]$/, 'Fraction', '¼'],
    ];
    world.model.two.regexNormal =
      world.model.two.regexNormal.concat(additionalTags);
  },
}

// Post-process hook
const plugin = {
  compute: {
    postProcess: doc => {
      doc.match('/[\u00BC-\u00BE\u2150-\u215E]/').tag('Fraction'); // support unicode fractions
    },
  },
  hooks: ['postProcess'],
};
```

**Comment by spencermountain** [comment, marked as answer]
this is very well done!!
Either should work - my only fear is that our match parser won't parse the second one properly - haha.
Bravo for getting so far mutating things, can I add this do the docs?
cheers

**Reply by track0x1** [reply]
Thank you! This one seems to work with the match parser, but with my more complex regex I switch to passing the JSON object with the regex instance `doc.match({ regex })` 👌 

Do you see the compromise internals (world internals) as public API equivalent to the other methods exposed (ie. match)? That was my only concern going with the mutate approach (if it is more likely to be changed without notice). Might be good to highlight in docs if either is ok or ill-advised.

**Comment by spencermountain** [comment]
also - please make a pr! these unicode fractions are really fun stuff

**Reply by track0x1** [reply]
I’d be happy to do so!
